test/integration/features/profile.feature

```diff
@@ -2,104 +2,161 @@
 Feature: Profile
   As a user I can perform basic operations of Minishift with profile feature
 
-  Scenario: Starting Minishift with default profile
-     Given Minishift has state "Does Not Exist"
-      When executing "minishift start" succeeds
-      Then Minishift should have state "Running"
+  Scenario Outline: As user, I cannot create profile with blank profile name
+    When executing "minishift profile set <profilename>" fails
+    Then stderr should contain
+    """
+    A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+    """
 
-Scenario: User should be able to list default profile 'minishift'
-   Given Minishift has state "Running"
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
+  Examples: Empty profile name
+    | profilename |
+    | ''          |
+    | ' '         |
+    
+  Scenario Outline: As user, I cannot create profile with special character in profile name
+    When executing "minishift profile set <profilename>" fails
+    Then stderr should contain
     """
-    - minishift	Running		(Active)
+    Profile names must consist of alphanumeric characters only.
     """
-  Scenario: Getting default profile internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
 
-Scenario: Starting Minishift with profile foo
-    When executing "minishift start --profile foo" succeeds
-    Then Minishift should have state "Running"
+  Examples: Wrong profile names
+    | profilename |
+    | '.'         |
+    | '-'         |
+    | '#$'        |
+    | '_VM'       |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
 
-  Scenario: Getting profile foo internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    When executing "minishift profile set <profilename>" succeeds
```

Comment:
Wrong indent as "Given" is not used.

Reply:
@gbraad AFAIU Given is used for precondition check, as a user i can directly run profile set without any profile precondition and moreover its not a good idea to add ```Given``` clause in scenario outline.